lib/lotus/helpers/html_helper/empty_html_node.rb

```diff
@@ -0,0 +1,77 @@
+module Lotus
+  module Helpers
+    module HtmlHelper
+      # Empty HTML node
+      #
+      # @since x.x.x
+      # @api private
+      class EmptyHtmlNode
+        # Attributes separator
+        #
+        # @since x.x.x
+        # @api private
+        ATTRIBUTES_SEPARATOR = ' '.freeze
+
+        # Attributes escape
+        # This is for data-* attributes.
+        #
+        # @since x.x.x
+        # @api private
+        ATTRIBUTES_ESCAPE = {
+          '_' => '-'
+        }.freeze
+
+        # Attributes escape
+        # This is for data-* attributes.
+        #
+        # @since x.x.x
+        # @api private
+        #
+        # @see Lotus::Helpers::HtmlHelper::EmptyHtmlNode::ATTRIBUTES_ESCAPE
+        ATTRIBUTES_ESCAPE_PATTERN = Regexp.union(*ATTRIBUTES_ESCAPE.keys)
+
+        # Initialize a new empty HTML node
+        #
+        # @param name [Symbol,String] the name of the tag
+        # @param attributes [Hash,NilClass] the optional tag attributes
+        #
+        # @return [Lotus::Helpers::HtmlHelper::EmptyHtmlNode]
+        #
+        # @since x.x.x
+        # @api private
+        def initialize(name, attributes)
+          @name       = name
+          @attributes = attributes
+        end
+
+        # Resolve and return the output
+        #
+        # @return [String] the output
+        #
+        # @since x.x.x
+        # @api private
+        def to_s
+          %(<#{ @name }#{attributes}>)
+        end
+
+        private
+        # Resolve the attributes
+        #
+        # @return [String,NilClass] the tag attributes
+        #
+        # @since x.x.x
+        # @api private
+        def attributes
+          return if @attributes.nil?
+          result = [nil]
```

Comment:
In case attributes are not present, the output would be `<hr >` or `<div ></div>`. Notice the blank space between the last letter of the name and the closing tag character.